core/src/serve_repair.rs

```diff
@@ -729,7 +728,6 @@ impl ServeRepair {
                 "{}: Ignored received repair requests from ME: {}",
                 my_id, stats.err_self_repair,
             );
-            inc_new_counter_debug!("serve_repair-handle-repair--eq", stats.err_self_repair);
         }
```

Comment:
Should we just remove this whole block? The metric is reported as `err_self_repair` so the `warn!` also seems redundant.

Reply:
This makes sense, but I'm not sure that operators would configure log file monitoring and report warnings even if the validator appeared to be functioning normally, particularly if it was a transient issue.

Maybe we could have a pattern where the datapoint would normally be logged as `datapoint_info` but some conditions would cause the datapoint to be logged with `datapoint_warn`. Then the errors could be discovered from metrics. But maybe this isn't worth doing here without a uniform plan.

Reply:
> but I'm not sure that operators would configure log file monitoring and report warnings even if the validator appeared to be functioning normally, particularly if it was a transient issue

Agreed on this aspect. And this is probably true for anything that doesn't crash / significantly degrade a node. But on the off chance that someone is looking at their logs, it could help. And especially for someone not submitting metrics (ideally everyone would but out of control obviously).

> Maybe we could have a pattern where the datapoint would normally be logged as datapoint_info but some conditions would cause the datapoint to be logged with datapoint_warn. Then the errors could be discovered from metrics.

One item to note is that the concept of log level doesn't exist in the metrics database. The level is only relevant for a node locally deciding whether or not it should submit a point. Once the datapoint has been sent across to the database, the concept of severity doesn't exist.

But on a similar line of thinking, maybe what we want/need is a dashboard (and/or alerts) that monitor specific metrics that we know correspond to bad events. We kind of have this notion with the `panic` datapoint on the cluster telemetry dashboard, but unfortunately, there is a lot of noise in there from stuff like nodes boot-looping because of a full disk.

Reply:
> But maybe this isn't worth doing here without a uniform plan.

I'm kind of leaning this way. For example, I know we have some other places where we log and datapoint, such as:
https://github.com/solana-labs/solana/blob/a57fe0c23d15f47af038556bc4249bdaf00143b8/runtime/src/bank.rs#L3191-L3202

Reply:
> One item to note is that the concept of log level doesn't exist in the metrics database. The level is only relevant for a node locally deciding whether or not it should submit a point. Once the datapoint has been sent across to the database, the concept of severity doesn't exist.

Yeah, I was suggesting this just so the datapoint would be submitted. It would still be necessary to know which specific metrics to look for, in this case `err_self_repair > 0`.

> But on a similar line of thinking, maybe what we want/need is a dashboard (and/or alerts) that monitor specific metrics that we know correspond to bad events. We kind of have this notion with the `panic` datapoint on the cluster telemetry dashboard, but unfortunately, there is a lot of noise in there from stuff like nodes boot-looping because of a full disk.

Right, if we just log the datapoint it'll probably still be lost in the noise (in a different location) unless it is being actively monitored. It would be interesting to explore how to do this effectively.

Reply:
> Yeah, I was suggesting this just so the datapoint would be submitted

Ahh, I see. Like a hypothetical situation where someone is running at `warn`, but if there is bad content in the datapoint, raising it from `info` to `warn` so that it does get emitted.